### DISCUSSION TITLE
No Audio From Mid-Slider Hitsounds In Lazer

### DISCUSSION THREAD

**Discussion by OnlyJustEthan** [ppy/osu#29621] [Q&A]
When playing through [Bad Apple!!](https://osu.ppy.sh/beatmapsets/751222#osu/3399434) on either difficulty, mid-slider sounds do not seem to play in the editor or in normal gameplay. These would be sounds seeming to come from a slider tick. Tail sound effects will still play, both at the end of a slider and when bouncing back to go the other direction. The sounds will play normally in the Stable editor and in normal gameplay. No video for this, but I tried the hardcoded Lazer skins and got the same result.

I do need to disclaim that I'm not super familiar with the editor, but I spent a while in both and could not figure out how those sounds were added, or how to make them work.

Example in Stable:

https://github.com/user-attachments/assets/19ba3e54-6142-4c33-a18f-5ebf9708aaae

Example in Lazer:

https://github.com/user-attachments/assets/56b04138-8bbe-4945-a4e4-28ba07ed6e07

**Comment by dotremilie** [comment]
This happens because stable editor let's you put multiple inherited timing points that change the custom sampleset of hitsounds. Lazer (for some reason) doesn't support that and ties hitsounding to the object which results in simplifying slider hitsounds to circle, body, reverse (if viable) and slider-end (the pink labels under these parts of sliders). so currently you cannot have multiple samplesets (a.k.a. use multiple files) for slider body. This also makes it so you can't lower or increase the volume of object (e.g. spinner or long slider) overtime.

What happened is mapper put 2 files for slider-tick and 2 for slider-reverse under the same sampleset number, one with actual sound and one empty, I'll give you timestamps of the timing points and try to be as accurate as possible but I don't see the map right now, just video.

`01:52:141` - Initial timing point for the 1st slider. It gives the slider the "muted" sampleset. To confirm that we have a tick near `01:52:571` that is indeed muted.
`01:53:006` - The timing point that changes the sampleset to actual sound. Indeed the sound is being played in stable, but not in Lazer.
`01:53:106` - Change of sampleset to the muted one again, so every tick that happens after is muted. I believe these 2 are omitted by Lazer,
`01:53:771` - Timing point that sets the volume to 5%, therefore any sound that is played is barely visible. Mappers do that when they put slider-end on some place (mostly blue-tick) where there is no distinct sound that would represent the slider-end.

Then there's the 2nd slider and the story repeats. The 3rd one has slider-reverse instead so Lazer reads `01:56:471` timing point as slider-reverse hitsound. This should dispel your doubts.

**Reply by OnlyJustEthan** [reply]
Thank you for the very detailed explaination! I couldn't find any issue reports mentioning this, is it something that I should bring up as an issue or wait an let other things be resolved first?

**Reply by dotremilie** [reply]
I believe this is what #24997 is about so no need to create a new issue.
